### ISSUE TITLE
Popup.pollPopup immediately closes the popup if the authorization server and the JS app share the same address

### ISSUE THREAD

**Issue by PinpointTownes** [sahat/satellizer#711]
*Moved from https://github.com/openiddict/core/issues/49#issuecomment-172234292*

_____

`Popup.pollPopup` immediately closes the popup if the authorization server and the JS app share the same address (which is a common case when using your own authorization server) without even checking if the current address of the popup matches the `redirect_uri` registered in the options.

```js
Popup.pollPopup = function() {
  var deferred = $q.defer();

  var polling = $interval(function() {
    try {
      var documentOrigin = document.location.host;
      var popupWindowOrigin = Popup.popupWindow.location.host;

      // This check will always be true if the authorization server and the JS frontend are served at the same address.
      if (popupWindowOrigin === documentOrigin && (Popup.popupWindow.location.search || Popup.popupWindow.location.hash)) {
        var queryParams = Popup.popupWindow.location.search.substring(1).replace(/\/$/, '');
        var hashParams = Popup.popupWindow.location.hash.substring(1).replace(/[\/$]/, '');
        var hash = utils.parseQueryString(hashParams);
        var qs = utils.parseQueryString(queryParams);

        angular.extend(qs, hash);

        if (qs.error) {
          deferred.reject(qs);
        } else {
          deferred.resolve(qs);
        }

        $interval.cancel(polling);

        Popup.popupWindow.close();
      }
    } catch (error) {
      // Ignore DOMException: Blocked a frame with origin from accessing a cross-origin frame.
    }

    if (!Popup.popupWindow || Popup.popupWindow.closed || Popup.popupWindow.closed === undefined) {
      $interval.cancel(polling);
    }
  }, 20);

  return deferred.promise;
};
```

/cc @Gillardo

**Comment by sahat** [comment]
@PinpointTownes @Gillardo Was that Satellizer PR (https://github.com/sahat/satellizer/pull/712) created for this issue?

**Comment by Gillardo** [comment]
Yes, sorry i mustof typed wrong issue number.  I have renamed PR

**Comment by Gillardo** [comment]
@sahat I have resubmitted with a new PR now

**Comment by sahat** [comment]
Fixed in #718.
